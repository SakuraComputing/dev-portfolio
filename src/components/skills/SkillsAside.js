import React from 'react';
import Skills from './Skills';

class SkillsAside extends React.Component {

    render() {

        const { skillsaside } = this.props;

        let skillItems;

        if(skillsaside.length > 0) {
            skillItems = skillsaside.map(skill => (
                <Skills key={skill.id} skill={skill}/>
            ))
        } else {
            skillItems = <h4>No skillsaside found....... </h4>
        }

        return (
                <aside className="skills aside section">
                    <div className="section-inner">
                        <h2 className="heading">Skills</h2>
                        <div className="content">
                            <p className="intro">
                                Intro about your skills goes here. Keep the list lean and only show your primary skillset. You can
                                always provide a link to your Linkedin or Coderwall profile so people can get more info there.</p>
                            <div className="skillset">
                                {skillItems}
                            </div>
                            <p><a className="more-link" href="#"><i className="fa fa-external-link"></i> More on Coderwall</a>
                            </p>
                        </div>
                    </div>
                </aside>

        );
    }
}
export default SkillsAside;


