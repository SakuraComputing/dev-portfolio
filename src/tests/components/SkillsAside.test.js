import React from 'react';
import { shallow } from 'enzyme';
import SkillsAside from '../../components/SkillsAside';

describe('SkillsAside Section', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<SkillsAside />);
        expect(wrapper).toMatchSnapshot();
    });
});