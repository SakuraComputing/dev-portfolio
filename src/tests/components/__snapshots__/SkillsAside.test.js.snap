// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillsAside Section should render correctly 1`] = `
<aside
  className="skills aside section"
>
  <div
    className="section-inner"
  >
    <h2
      className="heading"
    >
      Skills
    </h2>
    <div
      className="content"
    >
      <p
        className="intro"
      >
        Intro about your skills goes here. Keep the list lean and only show your primary skillset. You can always provide a link to your Linkedin or Coderwall profile so people can get more info there.
      </p>
      <div
        className="skillset"
      >
        <div
          className="item"
        >
          <h3
            className="level-title"
          >
            Python & Django
            <span
              className="level-label"
              data-animation="true"
              data-placement="left"
              data-toggle="tooltip"
              title="You can use the tooltip to add more info..."
            >
              Expert
            </span>
          </h3>
          <div
            className="level-bar"
          >
            <div
              className="level-bar-inner"
              data-level="96%"
            />
          </div>
        </div>
        <div
          className="item"
        >
          <h3
            className="level-title"
          >
            Javascript & jQuery
            <span
              className="level-label"
            >
              Expert
            </span>
          </h3>
          <div
            className="level-bar"
          >
            <div
              className="level-bar-inner"
              data-level="96%"
            />
          </div>
        </div>
        <div
          className="item"
        >
          <h3
            className="level-title"
          >
            HTML5, CSS3, SASS & LESS
            <span
              className="level-label"
            >
              Expert
            </span>
          </h3>
          <div
            className="level-bar"
          >
            <div
              className="level-bar-inner"
              data-level="96%"
            />
          </div>
        </div>
        <div
          className="item"
        >
          <h3
            className="level-title"
          >
            Ruby on Rails
            <span
              className="level-label"
            >
              Pro
            </span>
          </h3>
          <div
            className="level-bar"
          >
            <div
              className="level-bar-inner"
              data-level="85%"
            />
          </div>
        </div>
        <p>
          <a
            className="more-link"
            href="#"
          >
            <i
              className="fa fa-external-link"
            />
             More on Coderwall
          </a>
        </p>
      </div>
    </div>
  </div>
</aside>
`;
